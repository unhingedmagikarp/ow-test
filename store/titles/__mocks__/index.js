export default {
  namespaced: true,
  state: jest.fn().mockReturnValue(() => ({
    data: {},
  })),
  actions: {
    fetchTitles: jest.fn().mockResolvedValue(Promise.resolve()),
  },
  getters: {
    titles: jest.fn().mockReturnValue([]),
    getTitle: jest.fn().mockReturnValue({}),
  },
  mutations: {
    SET_TITLES: jest.fn(),
  },
}
