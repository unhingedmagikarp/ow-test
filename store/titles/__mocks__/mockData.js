export const MOCK_RESPONSE = {
  data: [
    {
      'Title Number': '342999',
      'Property Address': '102-108 Clerkenwell Road, London (EC1M 5SA)',
      Tenure: 'Freehold',
      X: -0.104842377,
      Y: 51.52256431,
    },
    {
      'Title Number': '41229',
      'Property Address': '75 Farringdon Road, London (EC1M 3JY)',
      Tenure: 'Freehold',
      X: -0.106601162,
      Y: 51.52122929,
    },
  ],
}

export const MOCK_STATE = {
  data: {
    "342999": MOCK_RESPONSE.data[0],
    "41229": MOCK_RESPONSE.data[1],
  },
}
