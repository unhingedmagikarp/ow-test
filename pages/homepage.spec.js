import Vue from 'vue'
import Vuex, { Store } from 'vuex'
import { shallowMount } from '@vue/test-utils'

import Homepage from './index'
import titlesStore from '@/store/titles'
import { MOCK_RESPONSE } from '~/store/titles/__mocks__/mockData'

jest.mock('@/store/titles')

Vue.use(Vuex)

describe('Page: HomePage', () => {
  let store

  titlesStore.getters.titles.mockReturnValue(MOCK_RESPONSE.data)

  beforeEach(() => {
    store = new Store({
      modules: {
        titles: {
          ...titlesStore,
        },
      },
    })
  })

  const defaultOpts = {
    mocks: {
      $route: { params: {}, query: {} },
    },
  }

  const factory = (opts = {}) =>
    shallowMount(Homepage, {
      store,
      ...defaultOpts,
      ...opts,
    })

  describe('matches snapshot', () => {
    it('when data is present', () => {
      const wrapper = factory()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  // TODO: test table functionality, missing data, data fetching
})
