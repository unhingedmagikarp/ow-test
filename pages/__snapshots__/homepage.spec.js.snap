// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: HomePage matches snapshot when data is present 1`] = `
"<client-only-stub>
  <div class="wrapper">
    <div>
      <v-table-stub data="[object Object],[object Object]" current-page="1" page-size="5">
        <thead>
          <v-th-stub sort-key="Title Number">Title Number</v-th-stub>
          <th>Class of Title</th>
        </thead>
      </v-table-stub>
      <smart-pagination-stub current-page="1" total-pages="0"></smart-pagination-stub>
    </div>
  </div>
</client-only-stub>"
`;
